src/layouts/dodge.js

```diff
@@ -39,8 +39,8 @@ export function dodgeY(dodgeOptions = {}, options = {}) {
 }
 
 function dodge(y, x, anchor, padding, options) {
-  const [, r] = maybeNumberChannel(options.r, 3);
-  return layout(options, (I, scales, {[x]: X, r: R}, dimensions) => {
+  return layout(options, function(I, scales, {[x]: X, r: R}, dimensions) {
+    const r = R ? undefined : this.r !== undefined ? +this.r : options.r !== undefined ? +options.r : 3;
```

Comment:
```suggestion
    const r = R ? undefined : this.r !== undefined ? this.r : options.r !== undefined ? +options.r : 3;
```

I think this.r is necessarily a number?

Reply:
It is in the case of all the marks we’ve implemented, but theoretically it might not be for some other custom mark someone else implements. Similarly, it’s theoretically possible that the `r` channel is not numbers—and we don’t protect against that. So, sure, we can remove the `+` here if you like.